src/main/java/org/openmrs/isanteplus/performancedata/model/connection/Inserter.java

```diff
@@ -0,0 +1,38 @@
+package org.openmrs.isanteplus.performancedata.model.connection;
+
+import org.openmrs.isanteplus.performancedata.model.AbstractEntity;
+import org.openmrs.isanteplus.performancedata.options.GeneratorOptions;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import org.springframework.util.CollectionUtils;
+import java.util.List;
+
+public class Inserter {
+
+    private final MySqlConnector connector;
+
+    public Inserter(GeneratorOptions options) {
+        this.connector = new MySqlConnector(options.getDbLogin(), options.getDbPassword(),
+                options.getDbServer(), options.getDbPort());
+    }
+
+    public void insertEntities(List<AbstractEntity> entities) throws SQLException {
+            Connection connection = connector.getConnection();
```

Comment:
ideally close should be done in a finally block, or try with resource should be sued